File: encryption/encryption.java
import java.nio.charset.StandardCharsets;
import java.security.SecureRandom;
import java.util.Base64;
import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import org.json.JSONObject;

/* 
 * This code depends on org.json library
 */

public class EncryptionUtils {


    public static String encrypt() {
        /**
        * This is the base64 encoded key from Console -> Settings -> API -> Encryption Key
        * Ideally, this should be stored in a secure location and not in the codebase. ex: env variable
        */
        String ENCRYPTION_KEY = "v4gaj6ELt8p5+aLuBeNbRW45BAkHQzAfC2sExq1Elr4=";
        byte[] key = Base64.getDecoder().decode(ENCRYPTION_KEY);

        JSONObject data = new JSONObject();
        data.put("timestamp", System.currentTimeMillis() / 1000); // Current UNIX timestamp in seconds
        data.put("page-id", "my-page-id-3");

        String json = data.toString();

        // Generate a random IV (Initialization Vector) for each encryption
        byte[] iv = new byte[16];
        new SecureRandom().nextBytes(iv);

        try {
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            SecretKeySpec secretKeySpec = new SecretKeySpec(key, "AES");
            IvParameterSpec ivParameterSpec = new IvParameterSpec(iv);

            cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec, ivParameterSpec);
            byte[] encrypted = cipher.doFinal(json.getBytes(StandardCharsets.UTF_8));

            // Combine encrypted data and IV into a single string separated by ":"
            String encryptedData = Base64.getEncoder().encodeToString(encrypted);
            String ivString = Base64.getEncoder().encodeToString(iv);
            return encryptedData + ":" + ivString;
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

    public static void main(String[] args) {
        String encryptedData = encrypt();
        System.out.println(encryptedData);
    }
}

File: encryption/encryption.js
/**
 * This depends on the crypto-js library
 * See https://www.npmjs.com/package/crypto-js
 */

const CryptoJS = require("crypto-js");

function encrypt() {
    /**
     * This is the base64 encoded key from Console -> Settings -> API -> Encryption Key
     * Ideally, this should be stored in a secure location and not in the codebase. ex: env variable
     */
    const ENCRYPTION_KEY = "v4gaj6ELt8p5+aLuBeNbRW45BAkHQzAfC2sExq1Elr4=";
    const key = CryptoJS.enc.Base64.parse(ENCRYPTION_KEY);

    const data = {
        /**
         * Current UNIX timestamp in seconds
         * This is used to ensure that the request is not replayed
         */
        timestamp: Math.floor(Date.now() / 1000),

        /**
         * Below, you would set the other data you want to set
         * See our documentation what keys are required in each component
         */
        "page-id": "my-page-id-3",
    };

    /**
     * Convert the data to a JSON string
     */
    const json = JSON.stringify(data);

    /**
     * Generate a random IV (Initialization Vector) for each encryption
     * This is used to ensure that the same data encrypted multiple times will have different results
     */
    const iv = CryptoJS.lib.WordArray.random(16);

    /**
     * Encrypt the data
     */
    const encrypted = CryptoJS.AES.encrypt(json, key, {
        iv: iv,
    });

    /**
     * Finally, return the encrypted data (base64 encoded) and the IV (base64 encoded)
     * Connect the two with a :
     */
    return encrypted.toString() + ":" + iv.toString(CryptoJS.enc.Base64);
}

console.log(encrypt());
